squad/core/models.py

```diff
@@ -990,11 +990,15 @@ def create_or_update(cls, build):
         regressions = None
         fixes = None
 
-        previous_build = Build.objects.filter(
-            status__finished=True,
-            datetime__lt=build.datetime,
-            project=build.project,
-        ).order_by('datetime').last()
+        previous_status = ProjectStatus.objects.filter(baseline=build).last()
```

Comment:
I was thinking about this, and it's easier to expose my point with a use case:

b1 is any finished build
b2.status.baseline = b1
b3.status.baseline = b1
b4.status.baseline = b3

When processing `create_or_update` for status of b3, `ProjectStatus.objects.filter(baseline=b3).last()` would retrieve status of b4, which is ahead of b3. Am I missing something here?

I know that b4 wouldn't probably exist by the time b3 is being processed, so that query would return None most of the time.

Reply:
Maybe you're right. I need to think about that. Maybe I did the comparison wrong way around.

Reply:
I did this the wrong way around. Should be fixed now.